DB/pgsql.php

```diff
@@ -1110,6 +1127,29 @@ function _checkManip($query)
                 || parent::_checkManip($query));
     }
 
+    // }}}
+    // {{{ _resultId()
+
+    /**
+     * Returns a numeric value identifying the result, used for maintaining
+     * indexes. Required by migration from resources to 'PgSql\Result'
+     * types with PHP 8.1.
+     *
+     * @param mixed $result  the query result resource or PgSql\Result
+     *
+     * @return int
+     *
+     * @access protected
+     */
+    function _resultId($result)
+    {
+        return 
+            is_resource($result)
+                ? function_exists('get_resource_id')
+                    ? get_resource_id($result) 
+                    : (int)$result
+                : spl_object_id($result);
+     }
```

Comment:
I suspect precedence rules mean this all works correctly, but it would be clearer if you used parentheses around the inner expression.

(It would be even clearer if you used `if`/`else` blocks, but whether or not that matters is a decision for the primary maintainer.)